ismumchow/mood
# Clerk Implemented for Authentication and Middleware

Summary
This PR introduces the implementation of Clerk for both authentication and middleware purposes. We are transitioning from our previous custom-built authentication to use Clerk, which provides more robustness and scalability. This implementation also includes middleware functions for better request handling and validation.

Changes
Integrated Clerk as our new authentication service.
Added middleware functions for handling secure routes and request validation.
Refactored existing codebase to remove deprecated custom authentication methods.
Updated user schema to accommodate Clerk IDs.
Features
Clerk for User Authentication

Implemented Clerk's user authentication methods including sign-up, login, and logout.
Added UI components for Clerk-powered authentication.
Middleware Functions

Created middleware for Clerk to validate and handle incoming HTTP requests.
Utilized Clerk's JWT validation for secure routing.
Code Cleanup

Removed old, deprecated authentication methods from the codebase.
Refactored various parts of the code to improve readability and scalability.
Testing
Updated existing tests and added new ones to cover Clerk's authentication and the new middleware functions.
Added unit tests using Jest and Enzyme.
Ran end-to-end tests to ensure complete functionality.
How to Test
Pull this branch locally.
Run npm install to install Clerk dependencies.
Start the application and test the new authentication methods.
Ensure that all tests pass.
